include/net/rest_client.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file rest_client.h
+ *
+ * @defgroup rest_client REST client library
+ * @{
+ * @brief REST client.
+ *
+ * @details Provide REST client functionality.
+ */
+
+#ifndef REST_CLIENT_H__
+#define REST_CLIENT_H__
+
+#include <net/http_parser.h>
+
+/** @brief TLS is not used. */
+#define REST_CLIENT_SEC_TAG_NO_SEC -1
+
+/** @brief Use default TLS peer verification, i.e., TLS_PEER_VERIFY_REQUIRED. */
+#define REST_CLIENT_TLS_DEFAULT_PEER_VERIFY -1
+
+/** @brief REST client does a sckt connection */
+#define REST_CLIENT_SCKT_CONNECT -1
+
+/** @brief Common HTTP status codes */
+enum rest_client_http_status {
+	REST_CLIENT_HTTP_STATUS_OK = 200,
+	REST_CLIENT_HTTP_STATUS_BAD_REQ = 400,
+	REST_CLIENT_HTTP_STATUS_UNAUTH = 401,
+	REST_CLIENT_HTTP_STATUS_FORBIDDEN = 403,
+	REST_CLIENT_HTTP_STATUS_NOT_FOUND = 404,
+};
+
+/**
+ * @brief REST client context.
+ *
+ * @details Parameters and data for using the REST client API.
+ *          Request related parameters should be filled before calling @ref rest_client_request
+ *          and when the function returns, response related data can be read from this structure.
+ */
+struct rest_client_req_resp_context {
+	/* Request: ***********************************************************/
+
+	/** In: Socket identifier for the connection. default: REST_CLIENT_SCKT_CONNECT and
+	 *      library will make a new socket connection.
```

Comment:
Yep, you could add that text in different places and repeat the default value. I'm submitting this now:
```
	/** Socket identifier for the connection. When using the default value,
	 *  the library will open a new socket connection. Default: REST_CLIENT_SCKT_CONNECT.
	 */

```
Please suggest otherwise if preferred.